src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -189,6 +192,64 @@ def nodestore_event(self) -> Event | None:
                 )
         return self._nodestore_event
 
+    def load_span_serialized_performance_issues(
+        self, light: bool, snuba_params: ParamsType
+    ) -> None:
+        """Rewriting load_performance_issues from scratch so the logic is more independent"""
+        for group_id in self.event["issue.ids"]:
+            group = Group.objects.filter(id=group_id, project=self.event["project.id"]).first()
+            if group is None:
+                continue
+
+            unique_spans: set[str] = set()
+            start: float | None = None
+            end: float | None = None
+            for problem in self.event["issue_occurrences"]:
+                parent_span_ids = problem.evidence_data.get("parent_span_ids")
+                if parent_span_ids is not None:
+                    unique_spans = unique_spans.union(parent_span_ids)
+            span = list(unique_spans)
+            for event_span in self.event["occurrence_spans"]:
+                suspect_spans: list[str] = []
+                for problem in self.event["issue_occurrences"]:
+                    offender_span_ids = problem.evidence_data.get("offender_span_ids", [])
+                    if event_span.get("span_id") in offender_span_ids:
+                        start_timestamp = float(event_span.get("precise.start_ts"))
+                        if start is None:
+                            start = start_timestamp
+                        else:
+                            start = min(start, start_timestamp)
+                        end_timestamp = float(event_span.get("precise.finish_ts"))
+                        if end is None:
+                            end = end_timestamp
+                        else:
+                            end = max(end, end_timestamp)
+                        suspect_spans.append(event_span.get("span_id"))
+            # Logic for qualified_short_id is copied from property on the Group model
+            # to prevent an N+1 query from accessing project.slug everytime
+            qualified_short_id = None
+            project_slug = self.event["project"]
+            if group.short_id is not None:
+                qualified_short_id = f"{project_slug.upper()}-{base32_encode(group.short_id)}"
```

Comment:
So we know that this project slug is unique per this `TraceEvent`, maybe instead allow `qualified_short_id` on the Group model to be passed a project_id to skip n+1, and mention this on the model instead so people can benefit in the future?

Reply:
I'll do this in a follow up